docs/book/src/plugins/declarative-v1.md

```diff
@@ -0,0 +1,55 @@
+# Declarative Plugin
+
+The declarative plugin is for building declarative cluster operators with kubebuilder.
+Declarative operators provide a fast path to orchestrating Kubernetes deployments to enable domain experts to focus
+on their component instead of re-answering questions like 'How do I get this YAML into the cluster?' or 'How do I
+update it?'.  
+With the declarative plugin manifests are created under `channels/packages/<packagename>/<version>/manifest.yaml` 
+and they will be loaded and applied to the cluster. An example of the pattern can be found [here](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v3-addon)
+The `channels/stable` file declares the current stable version that will be used by default.
```

Comment:
```suggestion
## When should I use this plugin

- If you are looking to scaffold one or more [controller's][controller-runtime] following its partner ( See an e.g. of the reconcile method implemented with. [here][addon-v3-controller]) 
- Also, if you are looking to have the scaffolded files shipped inside of your Manager container then, it might a good option for your use case. See on the Dockerfile the declarative plugin works with channels to allow you to push the manifests. [More info][dockerfile-addon] 
```

Reply:
```
[dockerfile-addon]: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3-addon/Dockerfile#L16-L19
```

Reply:
I think I would like to keep line 4 to 6 here as a general introduction / overview of the plugins purpose - I would remove line 7 to 9 and add your suggested 'When to use it' section, would you be okay with that?

Reply:
`[here][addon-v3-controller]` is not defined - where do you want to link to here?

Reply:
> I think I would like to keep line 4 to 6 here as a general introduction / overview of the plugins purpose - I would remove line 7 to 9 and add your suggested 'When to use it' section, would you be okay with that?

Okay, never mind that, I just saw you suggested changes to the introduction as well - ignore the above comment